### PR TITLE
Debug system calls on osx

### PR DESCRIPTION
**Pull request by m-kuhn**
This will log any external calls invoked on osx (with level "debug").
I found it quite handy to have a look at what `pkgbuild` commands are actually executed.

### CODE REVIEW

File: constructor/osxpkg.py [hunk 1]
```diff
@@ -21,6 +21,13 @@
 logger = logging.getLogger(__name__)
 
 
+def check_call(args):
+    """
+    Execute a system process and debug the invocation
+    """
+    logger.debug(f"Executing: {' '.join(args)}")
```
Comment (jaimergp) [inline constructor/osxpkg.py:28]:
```suggestion
    logger.debug("Executing: %s", " ".join(args))
```

File: constructor/osxpkg.py [hunk 2]
```diff
@@ -21,6 +21,13 @@
 logger = logging.getLogger(__name__)
 
 
+def check_call(args):
```
Comment (jaimergp) [inline constructor/osxpkg.py:24]:
```suggestion
def explained_check_call(args):
```

**Review by jaimergp — COMMENTED** [review]
LGTM. Can you move the helper function to `constructor.utils` and import from there? Thanks!

File: constructor/osxpkg.py (hunk 1 above)
Comment (m-kuhn) [reply constructor/osxpkg.py:28]: Just for my understanding, is this for consistency with the existing codebase or is there a reason to stay python <3.6 compatible?

**Comment by jaimergp** [comment]
Looks like there's a missing import on `osxpkg`

File: constructor/utils.py [hunk 3]
```diff
@@ -12,12 +12,20 @@
...
 from ruamel import yaml
 
 logger = logging.getLogger(__name__)
 
 
+def explained_check_call(args):
+    """
+    Execute a system process and debug the invocation
+    """
+    logger.debug(f"Executing: {' '.join(args)}")
```
Comment (jaimergp) [inline constructor/utils.py:26]:
```suggestion
    logger.debug("Executing: %s", " ".join(args))
```

**Comment by jaimergp** [comment]
pre-commit.ci autofix

File: constructor/osxpkg.py (hunk 1 above)
Comment (jaimergp) [reply constructor/osxpkg.py]: Consistency and best practices with `logging`.

**Comment by jaimergp** [comment]
Your fork doesn't allow us to commit directly to the branch so pre-commit can't autofix, sorry. You'll have to address the requested changes locally.

**Review by jaimergp — APPROVED** [review]
Thank you!

**Pull request merged**